src/xunit.performance.api/Profilers/Etw/SimpleTraceEventParser.cs

```diff
@@ -0,0 +1,378 @@
+            {
+                if (source.EventsLost > 0)
+                    throw new Exception($"Events lost in trace '{scenarioInfo.EventLogFileName}'");
+
+                ////////////////////////////////////////////////////////////////
+                // Process data
+                var parser = new KernelTraceEventParser(source);
+                parser.ProcessStart += (ProcessTraceData obj) => {
+                    if (IsOurProcess(obj.ProcessID) || IsOurProcess(obj.ParentID))
+                    {
+                        var process = new Process(obj.ImageFileName, obj.ProcessID, obj.ParentID);
+                        process.LifeSpan.Start = obj.TimeStamp;
+                        processes.Add(process);
+                    }
+                };
+                parser.ProcessStop += (ProcessTraceData obj) => {
+                    if (IsOurProcess(obj.ProcessID))
+                        processes.Single(process => process.Id == obj.ProcessID).LifeSpan.End = obj.TimeStamp;
+                };
+
+                ////////////////////////////////////////////////////////////////
+                // Image/Module data
+                parser.ImageLoad += (ImageLoadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    var module = new Module(obj.FileName, obj.ImageChecksum) {
+                        AddressSpace = new AddressSpace(obj.ImageBase, obj.ImageSize)
+                    };
+                    module.LifeSpan.Start = obj.TimeStamp;
+
+                    process.Modules.Add(module);
+                };
+                parser.ImageUnload += (ImageLoadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    // Check if the unloaded module is on the list.
+                    // The module must be loaded, in the same address space, and same file name.
+                    var module = process.Modules
+                        .SingleOrDefault(m => obj.TimeStamp > m.LifeSpan.End && obj.Equal(m));
+                    if (module == null)
+                        return;
+                    module.LifeSpan.End = obj.TimeStamp;
+                };
+
+                ////////////////////////////////////////////////////////////////
+                // "ntoskrnl.exe" data
+                Func<ImageLoadTraceData, bool> IsNtoskrnlModule = (ImageLoadTraceData obj) => {
+                    return obj.ProcessID == 0 && obj.FileName.EndsWith("ntoskrnl.exe");
+                };
+                parser.ImageDCStart += (ImageLoadTraceData obj) => {
+                    if (IsNtoskrnlModule(obj))
+                    {
+                        if (tmpNtoskrnlModule == null)
```

Comment:
Calling this a `tmp` kernel module seems odd, since it exists for the duration of the trace. Maybe call it "existing" or "default" or something?